app/models/gate.rb

```diff
@@ -1,13 +1,16 @@
+# frozen_string_literal: true
+
 # NOTE: Gate = 改札機のイメージ
 class Gate < ApplicationRecord
   FARES = [150, 190].freeze
-
   validates :name, presence: true, uniqueness: true
   validates :station_number, presence: true, uniqueness: true
 
   scope :order_by_station_number, -> { order(:station_number) }
 
   def exit?(ticket)
-    true
+    moved_distance = (station_number - ticket.entered_gate.station_number).abs
+    max_distance = FARES.index(ticket.fare) + 1
+    !moved_distance.zero? && max_distance >= moved_distance
```

Comment:
否定の`!`を使うと脳内で一瞬条件を判定させなければいけないので、

```suggestion
    moved_distance.positive? && max_distance >= moved_distance
```

とするのもアリです。（とはいえこれは好みのレベル）
https://docs.ruby-lang.org/ja/latest/method/Numeric/i/positive=3f.html